events/roommessageevent.cpp

```diff
@@ -70,6 +70,18 @@ MessageEventContent::Base* RoomMessageEvent::content() const
     return d->content;
 }
 
+template <class ContentT>
+MessageEventContent::Base* make(const QJsonObject& json)
+{
+    return new ContentT(json);
+}
+
+template <class ContentT>
+MessageEventContent::Base* make2(const QJsonObject& json)
```

Comment:
Well, yeah, didn't like it either. The "proper" modern C++ way would be to make a couple of specializations for the two types but it's code duplication again, and rather dumb one. And yes, VideoContent is supposed to be like ImageContent and FileContent (see below).